subsys/logging/log_core.c

```diff
@@ -211,7 +211,7 @@ void z_log_vprintk(const char *fmt, va_list ap)
 static log_timestamp_t default_get_timestamp(void)
 {
 	return IS_ENABLED(CONFIG_LOG_TIMESTAMP_64BIT) ?
-		sys_clock_tick_get() : k_cycle_get_32();
+		k_cycle_get_64() : k_cycle_get_32();
```

Comment:
the reason why `k_cycle_get_64()` is not used is that it's not that widely implemented (I don't know why).
I tried on `nrf52` and `qemu cortex-m` and both don't implement it.

I tried following code:
```
	printk("%d %lld %lld \n", k_cycle_get_32(), k_cycle_get_64(), sys_clock_tick_get());
	k_msleep(1000);
	printk("%d %lld %lld\n", k_cycle_get_32(), k_cycle_get_64(), sys_clock_tick_get());
```

And `k_cycle_get_64()` is always 0 on both platforms I checked.

Reply:
Thanks for the review. I wasn't aware that `k_cycle_get_64()` is not implemented on many platforms.

Please see my revised pull request. Instead of using `k_cycle_get_64()`, the timestamp frequency is selected based on `CONFIG_LOG_TIMESTAMP_64BIT`.

`native_posix_64` is where I ran into the issue. It's 100 ticks per second and 1000000 cycles per second so I was seeing the log timestamp being off by a factor of 10000.